src/components/buttons/ButtonGroup.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { ReactNode } from 'react';
+
+export enum ButtonGroupAlign {
+  Left = 'left',
+  Right = 'right',
+}
+
+export interface ButtonGroupProps {
+  align?: ButtonGroupAlign;
+  children: ReactNode;
+  isCombined?: boolean;
+}
+
+export function ButtonGroup({
+  align = ButtonGroupAlign.Left,
+  children,
+  isCombined = false,
+}: Readonly<ButtonGroupProps>) {
+  if (!isCombined) {
+    return <StyledButtonGroup align={align}>{children}</StyledButtonGroup>;
+  }
+
+  return <StyledCombinedButtonGroup align={align}>{children}</StyledCombinedButtonGroup>;
+}
+
+const StyledButtonGroup = styled.span<{ align: ButtonGroupAlign }>`
+  align-items: center;
+  display: flex;
+  flex: 1 0 auto;
+  gap: var(--echoes-dimension-space-100);
+  justify-content: ${({ align }) => (align === ButtonGroupAlign.Left ? 'flex-start' : 'flex-end')};
+  width: 100%;
+`;
+
+const StyledCombinedButtonGroup = styled(StyledButtonGroup)`
+  gap: 0;
+
+  & button {
```

Comment:
If you rebase on the last version of my PR you can do `& ${ButtonStyled} {...` instead.